src/main/java/com/github/manolo8/darkbot/backpage/dispatch/DispatchData.java

```diff
@@ -0,0 +1,123 @@
+package com.github.manolo8.darkbot.backpage.dispatch;
+
+import java.util.LinkedHashMap;
+import java.util.Map;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+public class DispatchData {
+    private int permit, gateUnits, availableSlots, maxSlots;
+
+    private final DataBuilder dataBuilder = new DataBuilder();
+    private final Map<String, Retriever> retrievers = new LinkedHashMap<>();
+    private final Map<String, InProgress> progressSlots = new LinkedHashMap<>();
+
+    public int getPermit() {
+        return permit;
+    }
+
+    public void setPermit(int permit) {
+        this.permit = permit;
+    }
+
+    public int getGateUnits() {
+        return gateUnits;
+    }
+
+    public void setGateUnits(int gateUnits) {
+        this.gateUnits = gateUnits;
+    }
+
+    public int getAvailableSlots() {
+        return availableSlots;
+    }
+
+    public void setAvailableSlots(int availableSlots) {
+        this.availableSlots = availableSlots;
+    }
+
+    public int getMaxSlots() {
+        return maxSlots;
+    }
+
+    public void setMaxSlots(int maxSlots) {
+        this.maxSlots = maxSlots;
+    }
+
+    public Map<String, Retriever> getRetrievers() {
+        return retrievers;
+    }
+
+    public void parseRow(String str) {
+        if(!dataBuilder.buildRetriever(str)) return;
+        dataBuilder.buildInProgress(str);
+    }
+
+    public Map<String, InProgress> getInProgress() {
+        return progressSlots;
+    }
```

Comment:
```suggestion
    public Map<String, InProgress> getInProgress() {
        return progressSlots;
    }

    public void parseRow(String str) {
        if (dataBuilder.buildRetriever(str)) return;
        dataBuilder.buildInProgress(str);
    }
```

Probably best to swap the order so all getters are there first, then this parsing method.

Also as noted in the other comment, you need to remove the !, since buildRetriever returns true if it succeeds, which, if it does, you want to return and not do the later